src/player.h

```diff
@@ -594,6 +594,7 @@ class Player final : public Creature, public Cylinder
 		bool isImmune(ConditionType_t type) const final;
 		bool hasShield() const;
 		bool isAttackable() const final;
+		bool lastHitIsPlayer(Creature* lastHitCreature) const;
```

Comment:
Yes, you understood correctly. A free function is a non-member function, and if you make it static and only on the cpp it's not visible outside that file compilation unit, so you don't need to recompile if the function signature ever changes, etc.

Declare it just before the function it is used. A static lambda is also useful if it is used only by a single method.